src/context/BookmarksContext.js

```diff
@@ -0,0 +1,39 @@
+import { createContext, useState } from "react";
+import { getBookmarks, addBookmark, removeBookmark } from "../api";
+
+const BookmarksContext = createContext();
+function BookmarksProvider({ children }) {
+  const [bookmarks, setBookmarks] = useState([]);
+
+  const fetchBookmarks = async () => {
+    const data = await getBookmarks();
+    setBookmarks(data);
+  };
+
+  const createBookmark = async (recipe) => {
+    const data = await addBookmark(recipe);
+    setBookmarks([...bookmarks, data]);
+  };
+
+  const deleteBookmark = async (id) => {
+    await removeBookmark(id);
```

Comment:
I would avoid use different names: `delete` vs `remove`. Pick one and stick with it. You can do something like this to avoid conflict the names
```
import * as api from "../api";
...
await api.removeBookmark(id)
```